### PR TITLE
[APIM-Analytics]Fix health monitor alerts table

### PR DESCRIPTION
**Pull request by ruwiniwj**
## Purpose
In APIM Health Monitor alerts, a new alert is added to the specific alert table only if the same alert hadn't occurred previously within 10 mins duration. Joining ApiHealthMonitorAlertStreamTemp stream to ApimApiHealthMonitorAlert table results in only the first alert being added to the health alert table, the subsequent alerts gets dropped. This is fixed by joining ApiHealthMonitorAlertStreamTemp stream to itself to check whether the same alert has occurred within a window of 10 mins.

### CODE REVIEW

File: features/apim-analytics-feature/org.wso2.analytics.solutions.apim.analytics.feature/src/main/resources/siddhi-files/apim_health_availability_alert_0.siddhi [hunk 1]
```diff
@@ -84,7 +84,7 @@ having AAS.message is null
 insert into ApimAllAlertSummary;
 
 @info(name='Query inserts data to ApimApiHealthMonitorAler')
- from ApiHealthMonitorAlertStreamTemp#window.length(1) as AH left outer join ApimApiHealthMonitorAlert#window.time(10 minutes) as AM
+from ApiHealthMonitorAlertStreamTemp#window.length(1) as AH left outer join ApiHealthMonitorAlertStreamTemp#window.time(10 minutes) as AM
```
Comment (shilmyhasan) [inline features/apim-analytics-feature/org.wso2.analytics.solutions.apim.analytics.feature/src/main/resources/siddhi-files/apim_health_availability_alert_0.siddhi]: We can introduce a new in memory stream here - ApimApiHealthMonitorNewTempStream

Comment (ruwiniwj) [reply features/apim-analytics-feature/org.wso2.analytics.solutions.apim.analytics.feature/src/main/resources/siddhi-files/apim_health_availability_alert_0.siddhi]: Fixed

Comment (shilmyhasan) [reply features/apim-analytics-feature/org.wso2.analytics.solutions.apim.analytics.feature/src/main/resources/siddhi-files/apim_health_availability_alert_0.siddhi]: can we change the join command like this to make it more accurate "from ApiHealthMonitorAlertStreamTemp#window.length(1) as AH left outer join ApiHealthMonitorAlertStream#window.time(10 minutes) as AM"

**Comment by Niveathika92** [comment]
Pending Dev discussion: 
Mail Thread : "Change APIM Health Monitor alerts siddhi file"

File: features/apim-analytics-feature/org.wso2.analytics.solutions.apim.analytics.feature/src/main/resources/siddhi-files/apim_health_availability_alert_0.siddhi [hunk 2]
```diff
@@ -47,6 +47,10 @@ define stream EmailAlertStream (type string, message string, alertTimestamp	stri
 
 define stream ApimAllAlertSummary (type string, tenantDomain string, message string, severity int, alertTimestamp long);
 
+Define stream ApimApiHealthMonitorAlertStream (apiName string,apiVersion string,apiCreator string, apiCreatorTenantDomain string, message string, severity int, alertTimestamp long);
```
Comment (Niveathika92) [inline features/apim-analytics-feature/org.wso2.analytics.solutions.apim.analytics.feature/src/main/resources/siddhi-files/apim_health_availability_alert_0.siddhi]: @ruwiniwj , The defined stream is different from the one used in join query.

Comment (ruwiniwj) [reply features/apim-analytics-feature/org.wso2.analytics.solutions.apim.analytics.feature/src/main/resources/siddhi-files/apim_health_availability_alert_0.siddhi]: Fixed with https://github.com/wso2/analytics-solutions/pull/253

**Pull request merged**